content/en/docs/concepts/scheduling-eviction/dynamic-resource-allocation.md

```diff
@@ -0,0 +1,214 @@
+responsible for tracking and allocating resources. Different kinds of
+resources support arbitrary parameters for defining requirements and
+initialization.
+
+## {{% heading "prerequisites" %}}
+
+Kubernetes v{{< skew currentVersion >}} includes cluster-level API support for
+dynamic resource allocation, but it [needs to be
+enabled](#enabling-dynamic-resource-allocation) explicitly.  You also must
+install a resource driver for specific resources that are meant to be managed
+using this API.  If you are not running Kubernetes v{{< skew currentVersion>}},
+check the documentation for that version of Kubernetes.
+
+<!-- body -->
+
+## API
+
+The new `resource.k8s.io/v1alpha1` {{< glossary_tooltip text="API group"
+term_id="api-group" >}} provides four new types:
+
+ResourceClass
+: Defines which resource driver handles a certain kind of
+  resource and provides common parameters for it. ResourceClasses
+  are created by a cluster administrator when installing a resource
+  driver.
+
+ResourceClaim
+: Defines a particular resource instances that is required by a
+  workload. Created by a user (lifecycle managed manually, can be shared
+  between different Pods) or for individual Pods by the control plane based on
+  a ResourceClaimTemplate (automatic lifecycle, typically used by just one
+  Pod).
+
+ResourceClaimTemplate
+: Defines the spec and some meta data for creating
+  ResourceClaims. Created by a user when deploying a workload.
+
+PodScheduling
+: Used internally by the control plane and resource drivers
+  to coordinate pod scheduling when ResourceClaims need to be allocated
+  for a Pod.
+
+Parameters for ResourceClass and ResourceClaim are stored in separate objects,
+typically using the type defined by a {{< glossary_tooltip
+term_id="CustomResourceDefinition" text="CRDs" >}} that was created when
+installing a resource driver.
+
+The `core/v1` `PodSpec` defines ResourceClaims that are needed for a Pod in a new
+`resourceClaims` field. Entries in that list reference either a ResourceClaim
+or a ResourceClaimTemplate. When referencing a ResourceClaim, all Pods using
+this PodSpec (for example, inside a Deployment or StatefulSet) share the same
+ResourceClaim instance. When referencing a ResourceClaimTemplate, each Pod gets
+its own instance.
+
+The `resources.claims` list for container resources define whether a container gets
+access to these resource instances, which makes it possible to share resources
+between one or more containers.
+
+Here is an example for a fictional resource driver. Two ResourceClaim objects
+will get created for this Pod and each container gets access to one of them.
+
+```yaml
+apiVersion: resource.k8s.io/v1alpha1
+kind: ResourceClass
+name: resource.example.com
+driverName: resource-driver.example.com
+---
+apiVersion: cats.resource-driver.example.com/v1
+kind: ClaimParameters
+name: resource-claim-parameters
+color: black
+size: large
```

Comment:
You need a `spec` here.

Reply:
Does each CRD need a spec?

Reply:
I guess it's not strictly necessary, but I've never seen anyone *not* use one and any tutorials / examples you come across always includes them. If nothing else, it serves to group the custom fields associated with the CRD rather than having these custom fields right next to the standard / required ones. I don't feel too strongly about if for this example, but it was a bit jarring (at least for me) to see this CRD without one.

Reply:
I can buy the "group the custom fields" and "matches examples" argument. I'm just wondering whether we perpetuate some cargo-culting here by showing a "spec" sub-structured that isn't really needed. If we do it, all future resource driver CRDs will probably have it...

Anyway, I'll changed it.

Reply:
Done.